common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,34 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def has_path_to_root(xblock, course):
+            """ Check recursively if an xblock has a path to the course root """
+            xblock_parents = self._get_parents_from_structure(xblock, course.structure)
+
+            if len(xblock_parents) == 0 and xblock.type in ["course", "library"]:
```

Comment:
@cpennington  is there a "root class" or anything like that we can check against, instead of hardcoding this here?